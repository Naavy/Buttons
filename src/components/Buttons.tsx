import React, { FC, useContext, useState } from "react";
import styled from "styled-components";
import { ButtonsContext } from "../context/ButtonsContext";

const ButtonsContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 300px;
`;

const Button = styled.button`
  width: 70px;
  padding: 5px;
  margin: 10px;
  background-color: transparent;
  color: black;
  border-top: none;
  border-bottom: none;
  border-left: 1px #000 solid;
  border-right: 1px #000 solid;
  cursor: pointer;
`;

const Buttons: FC = () => {
  const { setFirstClass, setSecondClass, setThirdClass } =
    useContext(ButtonsContext);
  const [timer, setTimer] = useState<NodeJS.Timeout>();

  const numberHide = "number-hide";
  const numberAnimationDisplay = "number-animation-display";
  const numberAnimationShow = "number-animation-show";

  const showAllNumbers = () => {
    setTimer(
      setTimeout(() => {
        setFirstClass(numberAnimationDisplay);
        setSecondClass(numberAnimationDisplay);
        setThirdClass(numberAnimationDisplay);
      }, 4000)
    );
  };

  const onLeft = () => {
    setThirdClass(numberAnimationShow);
    setSecondClass(`${numberAnimationShow}-1`);
    setFirstClass(`${numberAnimationShow}-2`);
    showAllNumbers();
  };

  const onRight = () => {
    setFirstClass(numberAnimationShow);
    setSecondClass(`${numberAnimationShow}-1`);
    setThirdClass(`${numberAnimationShow}-2`);
    showAllNumbers();
  };

  const reset = () => {
    if (timer) clearTimeout(timer);
    setFirstClass(numberHide);
    setSecondClass(numberHide);
    setThirdClass(numberHide);
  };

  return (
    <ButtonsContainer data-testid="buttons">
      <div style={{ display: "flex", justifyContent: "space-around" }}>
        <Button onClick={onLeft}>w lewo</Button>
        <Button onClick={onRight}>w prawo</Button>
      </div>
      <div>
        <Button onClick={reset}>reset</Button>
      </div>
    </ButtonsContainer>
  );
};

export default Buttons;
