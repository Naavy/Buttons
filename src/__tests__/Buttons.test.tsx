import React from "react";
import { render, screen } from "@testing-library/react";

import Buttons from "../components/Buttons";

test("Search Loader component renders properly", () => {
  render(<Buttons />);

  // Buttons renders properly
  const ButtonsNode = screen.getByTestId("buttons");
  expect(ButtonsNode).toBeInTheDocument();

  expect(ButtonsNode).toMatchSnapshot();
});
