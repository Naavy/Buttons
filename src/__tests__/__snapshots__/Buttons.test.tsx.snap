// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Loader component renders properly 1`] = `
<div
  class="sc-bdfBQB bfCCdG"
  data-testid="buttons"
>
  <div
    style="display: flex; justify-content: space-around;"
  >
    <button
      class="sc-gsTEea kzhPZE"
    >
      w lewo
    </button>
    <button
      class="sc-gsTEea kzhPZE"
    >
      w prawo
    </button>
  </div>
  <div>
    <button
      class="sc-gsTEea kzhPZE"
    >
      reset
    </button>
  </div>
</div>
`;
